### PR TITLE
Fix automate process

### PR DESCRIPTION
**Pull request by Lee-W**
* fix release commit validation step failure
    * we'll need to quote the commit message when running the bash script
* change the regex to check the release commit to include the PR number
    * Our release message format is something like "Release 1.0.0 (#123)" instead of "Release 1.0.0" 
* use BOT_ACCESS_TOKEN to push the git tag
* remove bump dev version step for now as it will fail due to commitizen behavior

### CODE REVIEW

File: .github/scripts/validate_release_commit.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 import sys
 
 if __name__ == "__main__":
-    pattern = r"^Release \d{1,}\.\d{1,}\.\d{1,}$"
+    pattern = r"^Release \d{1,}\.\d{1,}\.\d{1,}( \(#\d+\)){0,1}$"
```
Comment (pankajkoti) [inline .github/scripts/validate_release_commit.py:5]: Can you please elaborate this? I did not understand this part. 

Looking at the PR description, it also has non-digit part showing in the description, but here we have only digits.

I am checking the previous release PR. Do we mean to say it is the PR number?

`Release 1.17.2 #1262`

If yes, it does not seem to have closing round brackets around the PR number hash. Do we need those brackets?

Screenshot 2023-08-07 at 12 33 45 PM

Comment (pankajkoti) [reply .github/scripts/validate_release_commit.py:5]: Screenshot 2023-08-07 at 12 37 02 PM
 This is how the PR description reads and it contains non-digit characters too.

Comment (sunank200) [reply .github/scripts/validate_release_commit.py:5]: +1

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -14,14 +14,14 @@ jobs:
...
-          token: ${{ secrets.GITHUB_TOKEN }}
+          token: ${{ secrets.BOT_ACCESS_TOKEN }}
 
       - name: Check head commit message
         id: check_head_commit_message
         continue-on-error: true
         run: |
           head_commit_message=`git log --format=%s -1`
-          python ./.github/scripts/validate_release_commit.py $head_commit_message
+          python ./.github/scripts/validate_release_commit.py "$head_commit_message"
```
Comment (sunank200) [inline .github/workflows/release.yaml:24]: why do we need the quotes?

File: .github/scripts/validate_release_commit.py (hunk 1 above)
Comment (Lee-W) [reply .github/scripts/validate_release_commit.py:5]: Yes, I think you're right. Let me fix it now

File: .github/workflows/release.yaml (hunk 2 above)
Comment (Lee-W) [reply .github/workflows/release.yaml:24]: Yes, I think this is one of the main reason why it fails last time. It seems to be passed as `python ./.github/scripts/validate_release_commit.py Release 1.17.2 #123123` which causes an error

File: .github/scripts/validate_release_commit.py (hunk 1 above)
Comment (Lee-W) [reply .github/scripts/validate_release_commit.py]: Just pushed the fix

**Review by pankajkoti — APPROVED** [review]

File: .github/scripts/validate_release_commit.py (hunk 1 above)
Comment (Lee-W) [reply .github/scripts/validate_release_commit.py]: @pankajkoti @sunank200  I just checked again. We actually need those brackets. We're checking the commit message. not the PR title
圖片

Comment (pankajkoti) [reply .github/scripts/validate_release_commit.py]: okay @Lee-W

We can merge and test with what you feel works :)

Comment (Lee-W) [reply .github/scripts/validate_release_commit.py:5]: Thanks! Let me merge it now.

**Pull request merged**